packages/toolpad-utils/src/fs.ts

```diff
@@ -52,8 +65,9 @@ export async function writeFileRecursive(
 
 export async function updateYamlFile(filePath: string, content: object) {
   const oldContent = await readMaybeFile(filePath);
-  const newContent = oldContent ? yamlOverwrite(oldContent, content) : yaml.stringify(content);
 
+  let newContent = oldContent ? yamlOverwrite(oldContent, content) : yaml.stringify(content);
+  newContent = await formatYaml(newContent!, filePath);
```

Comment:
`newContent` should always be a string. We want to avoid using the no-null-assertion operator.

```suggestion
  newContent = await formatYaml(newContent, filePath);
```